trace-sdk/build.gradle

```diff
@@ -32,6 +33,92 @@ android {
+    android.'libraryVariants'
+            .all { variant ->
+                def variantName = variant.name
+                def unitTestTask = "test${variantName.capitalize()}UnitTest"
+                def androidTestCoverageTask = "create${variantName.capitalize()}CoverageReport"
+
+                // Add the report task name for the given variant to the Set above
+                reportTaskNames.add("create${variantName.capitalize()}AndroidTestCoverageReport")
+
+                tasks.create(name: "${unitTestTask}Coverage", type: JacocoReport, dependsOn: [
+                        "$unitTestTask",
+                        "$androidTestCoverageTask"
+                ]) {
+                    group = "Reporting"
+                    description = "Generate Jacoco coverage reports for the ${variantName.capitalize()} build"
+
+                    reports {
+                        html.enabled = true
+                        xml.enabled = true
+                        csv.enabled = true
+                    }
+
+                    def excludes = [
+                            '**/R.class',
+                            '**/R$*.class',
+                            '**/BuildConfig.*',
+                            '**/Manifest*.*',
+                            '**/*Test*.*',
+                            'android/**/*.*',
+                    ]
+
+                    def javaClasses = fileTree(dir: variant.javaCompileProvider.get().destinationDir,
+                            excludes: excludes)
+
+                    classDirectories.setFrom(files([
+                            javaClasses
+                    ]))
+
+                    def variantSourceSets = variant.sourceSets.java.srcDirs.collect { it.path }.flatten()
+                    sourceDirectories.setFrom(project.files(variantSourceSets))
+
+                    def androidTestsData = fileTree(dir: "${buildDir}/outputs/code_coverage/${variantName}AndroidTest/connected/", includes: ["**/*.ec"])
+
+                    executionData(files([
+                            "$project.buildDir/jacoco/${unitTestTask}.exec",
+                            androidTestsData
+                    ]))
+                }
+            }
+    // Get the task of running the connected tests
+    Task connectedAndroidTestTask = tasks.findByName("connectedAndroidTest")
+    // Create a set that will contain the required dependsOn elements
+    Set<?> deps = new HashSet<>();
+    connectedAndroidTestTask.dependsOn.each {
+        // Filter out Jacoco report tasks. this is an issue, connected tests should not depend on them
+        String providerName = ((TaskProvider) it).getName()
+        if (!reportTaskNames.grep{it == providerName}) {
+            deps.add(it)
+        }
+    }
+    // Replace the original dependsOn with the filtered
+    connectedAndroidTestTask.setDependsOn(deps)
+
+    // Finally make sure that each report task depends on the connected tests.
```

Comment:
this looks like it's failing the build as it's trying to run the connected tests as well i'm not sure this is accurate - we run the unit tests in the build-trace_sdk workflow and this now fails.